containers/runtime/driver/run.sh

```diff
@@ -35,7 +35,7 @@ if [ -n "${BQ_RESULT_TABLE}" ]; then
   fi
   if [ -r "${NODE_INFO_OUTPUT_FILE}" ]; then
     cp "${NODE_INFO_OUTPUT_FILE}" node_info.json
-    if [ -n "${SERVER_TARGET_OVERRIDE}" ]; then
+    if  [ "$(dig +short -t srv prometheus.test-infra-system.svc.cluster.local)" ] && [ -z "${DISABLE_PROMETHEUS}" ]; then
```

Comment:
On second thoughts, it may be better to test if SERVER_TARGET_OVERRIDE or ENABLE_PROMETHEUS are set. So Prometheus is used by default for PSM tests, and can be enabled with the annotation for other tests.